api/src/lib/endpoints/book.rs

```diff
@@ -8,8 +9,12 @@ use self::models::*;
 use crate::middleware::session::Session;
 use crate::response;
 
-pub async fn get_book_by_id(_: Session, id: u32) -> Result<impl Reply, Infallible> {
-    if let Ok(mut client) = crate::rpc::book_client().await {
+pub async fn get_book_by_id(
+    _: Session,
+    addr: SocketAddr,
+    id: u32,
+) -> Result<impl Reply, Infallible> {
+    if let Ok(mut client) = crate::rpc::book_client(&addr).await {
```

Comment:
Take the `book_client` from book service library. Please duplicate to all book endpoints.